docs/planning/sprintplanning.md

```diff
@@ -0,0 +1,66 @@
+- Story/Feature typically spans 1-2 weeks for one engineer.
+To propose a new feature,  create a new issue of type “Feature Request”
+- Please fill in as much detail as possible in the Feature template and propose a release cycle
+- All new features will be labelled “kind/feature”, and “triage/needstriage”
+- All new proposals are discussed either in WG meeting, or on slack, or dedicated meetings if needed at any point scheduled by the proposer. 
+- By R-2w’s WG meeting all new proposals should be vetted and verified that they fit the [Planning criteria](#planning-criteria). Features that do not fit this criteria will not be considered in the current release planning cycle and hence early discussion is encouraged instead of waiting for R-2w.
+- All eligible proposals will then be snapshotted into the voting document and open for voting till R-1w WG meeting. In R-1w WG meeting top features based on votes will be planned for next release. 
+> We need some kind of estimates for each feature for release planning. Shall we adopt story points on fibonacci scale or t-shirt sizing? My vote goes to story points on fibonacci scale to avoid spending a lot of time on estimation, which is a common pitfall.
+- Selected features will be labeled “triage/accepted”. Rest will be labelled “triage/ice-boxed” and will be eligible for next planning cycle. Accepted features will be added to the correct milestone and sprint specific project for tracking.
+
+## How to manage Epic/Scenario?
+- Epic/Scenario typically spans 1-3 months or even more in some cases.
+- For new proposals these will be created and triaged/refined just like Story/Feature. However, after triaging these need to be broken down into Features for planning and voting.
+- Github doesn’t allow nested Issues, hence the only good way to track these are by creating Projects and adding the broken down features as issues to this project. An issue can be part of multiple projects and hence we can get a Scenario\Epic view as well as a Sprint view through respective projects.
+
+## Planning criteria:
+- Features are well defined and scoped with clear exit criteria. No feature creeps once release starts
+- Features can be implemented in <2w by one engineer. In case of Epic/Scenario, it could be broken down into <2w features with clear exit criteria.
+- The engineering team should be confident that any engineer can implement the feature based on the content in the feature *(This is similar to refinement in a scrum, but doesn’t have to be as strict as a functional specification)*.
+
+## Organizing with Github Projects 
+- New Project per sprint. Today, the sprint and milestone are both aligned. However in future we may decide to have multiple sprints in a release cycle.
+- New Project per any Story/Scenario
+- Any other projects that help organize and will be created when needed.
+
+## Organizing with Github Milestone 
+- A new milestone is created in Github for each 6 weeks release cycle.
```

Comment:
We might also have milestones for point releases that happen between the 6-week cycle (every week, currently)

Reply:
Are these planned weekly releases or these are out-of-band releases?
I thought the 6 weeks one is the only planned one, however there could be more releases. And we can create milestones for them if we need. Sorry, didn't understand the suggestion here :)

Reply:
I think point releases happen automatically when we commit to the release branch. I don't recall if this is weekly or more often, but do you think we need a milestone to track these PRs?

Reply:
I believe automatic point release is every Tuesday.

Here's the "user story" I'm thinking of: I'm release coordinator. I want to see all PRs that need to be backported to release `N-1`. How do I find them?

Thinking about it more carefully, we can't use milestones for this because issues can only have one milestone. Instead we'd probably want some kind of `backport/release-X.Y` labeling system.

Reply:
I'd suggest that we should backport the PRs as soon as they merge into master (possibly as a follow-on issue if we actually need to resolve merge conflicts or don't have correct permissions). As such, we shouldn't need a release coordinator beyond the "should we do it" question, which shouldn't require much blocking.